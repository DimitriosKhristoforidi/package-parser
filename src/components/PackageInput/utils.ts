import API from '../../API';
import { Modal } from 'antd';
import { TGetPackagesData } from './PackageInputTypes';

export const getPackagesData: TGetPackagesData = async (dependenciesList) => {
  if (!dependenciesList.length) {
    Modal.error({ title: 'JSON is empty' });
    return [];
  }

  const promises = dependenciesList.map((dep) => API.getPackage(dep));

  return await Promise.all(promises).then((res) =>
    res.map((item) => item.results[0].package),
  );
};
