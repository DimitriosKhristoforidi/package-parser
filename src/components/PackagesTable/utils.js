import { Modal, Tag } from 'antd';
import API from '../../API';

export const columns = [
  {
    title: 'Title',
    dataIndex: 'name',
    key: 'name',
    width: '15%',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    width: '40%',
  },
  {
    title: 'Tags',
    dataIndex: 'keywords',
    key: 'keywords',
    render: keywords => (
      <>
        {
          keywords?.map(tag =>
            <Tag color="blue" key={tag}>
              {tag.toUpperCase()}
            </Tag>,
          )
        }
      </>
    ),
  },
  {
    title: 'NPM link',
    dataIndex: 'links',
    key: 'links',
    render: links => (
      <a target="_blank" rel="noreferrer" href={links.npm}>{links.npm}</a>
    ),
  },
];

export const getPackagesData = async (list) => {
  if (list.length) {
    const promises = [];
    for (let i in list)
      promises.push(API.getPackage(list[i]));
    return await Promise.all(promises)
      .then(r => r.map(item => item.results[0].package));
  } else {
    Modal.error({ title: 'JSON is empty' });
  }
};
