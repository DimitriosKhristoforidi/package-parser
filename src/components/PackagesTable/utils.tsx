import React from 'react';
import { Tag } from 'antd';

export const columns = [
  {
    title: 'Title',
    dataIndex: 'name',
    key: 'name',
    width: '15%',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    width: '40%',
  },
  {
    title: 'Tags',
    dataIndex: 'keywords',
    key: 'keywords',
    render: (keywords: string[]) => (
      <>
        {keywords?.map((keyword, id) => (
          <Tag color='blue' key={`${keyword}-${id}`}>
            {keyword.toUpperCase()}
          </Tag>
        ))}
      </>
    ),
  },
  {
    title: 'NPM link',
    dataIndex: 'links',
    key: 'links',
    render: (links: { npm: string }) => (
      <a target='_blank' rel='noreferrer' href={links.npm}>
        {links.npm}
      </a>
    ),
  },
];
