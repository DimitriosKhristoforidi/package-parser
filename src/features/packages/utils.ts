import {
  TGetEmptyPackage,
  TGetPackagesData,
  TIsArray,
  TIsEmpty,
  TValidatePackagesTerm,
} from './packagesSliceTypes';
import { EMPTY_JSON, ERROR_MESSAGES } from './constants';
import API from '../../api';

export const getEmptyPackage: TGetEmptyPackage = (name) => {
  return {
    name,
    description: '',
    author: { name: '' },
    maintainers: [],
    homepage: '',
  };
};

export const getPackagesData: TGetPackagesData = async (dependenciesList) => {
  const promises = dependenciesList.map((dep) => API.getPackage(dep));

  return await Promise.all(promises);
};

export const isArray: TIsArray = (array) => {
  return Array.isArray(array);
};

export const isEmpty: TIsEmpty = (line) => {
  return !line || line === EMPTY_JSON;
};

export const validatePackagesTerm: TValidatePackagesTerm = async (
  packagesTerm,
) => {
  if (isEmpty(packagesTerm)) {
    throw new Error(ERROR_MESSAGES.EMPTY_JSON);
  }
  try {
    const json = JSON.parse(packagesTerm);
    if (isArray(json)) {
      throw new Error();
    }
  } catch {
    throw new Error(ERROR_MESSAGES.NOT_A_JSON);
  }
};
