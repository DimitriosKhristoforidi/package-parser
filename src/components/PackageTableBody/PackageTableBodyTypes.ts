import { IPackage } from '../../features/packages/packagesSliceTypes';

export interface IPackageTableBody {
  packagesList: IPackage[];
}
