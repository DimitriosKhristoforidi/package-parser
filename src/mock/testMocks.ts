import { RootState } from '../app/types';
import { initialState } from '../features/packages/packagesSlice';

export const packagesListMock = [
  {
    name: '@reduxjs/toolkit',
    description:
      'The official, opinionated, batteries-included toolset for efficient Redux development',
    maintainers: [
      {
        name: 'acemarke',
        email: 'mark.erikson@gmail.com',
      },
      {
        name: 'timdorr',
        email: 'timdorr@timdorr.com',
      },
      {
        name: 'gaearon',
        email: 'dan.abramov@gmail.com',
      },
      {
        name: 'phryneas',
        email: 'mail@lenzw.de',
      },
    ],
    homepage: 'https://redux-toolkit.js.org',
  },
];

export const initialStateMock: RootState = {
  packages: initialState,
};
