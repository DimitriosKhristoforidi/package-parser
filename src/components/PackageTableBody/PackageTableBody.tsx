import React, { useMemo } from 'react';
import { Chip, Grid, TableBody, TableCell, TableRow } from '@mui/material';
import { IPackageTableBody } from './PackageTableBodyTypes';

const PackageTableBody: React.FC<IPackageTableBody> = ({ packagesList }) => {
  const packagesListRender = useMemo(
    () =>
      packagesList.map((packageItem, idx) => (
        <TableRow key={idx}>
          <TableCell scope='row'>{packageItem.name}</TableCell>
          <TableCell>{packageItem.description}</TableCell>
          <TableCell>{packageItem.author?.name}</TableCell>
          <TableCell>
            <Grid container spacing={1}>
              {packageItem.maintainers.map((maintainer, idx) => (
                <Grid item key={idx}>
                  <Chip
                    color='primary'
                    variant='outlined'
                    label={maintainer.name}
                    size='small'
                  />
                </Grid>
              ))}
            </Grid>
          </TableCell>
          <TableCell align='right'>
            <a target='_blank' href={packageItem.homepage}>
              {packageItem.homepage}
            </a>
          </TableCell>
        </TableRow>
      )),
    [packagesList],
  );

  return <TableBody>{packagesListRender}</TableBody>;
};

export default PackageTableBody;
