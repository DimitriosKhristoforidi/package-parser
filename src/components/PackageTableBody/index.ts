import PackageTableBody from './PackageTableBody';

export default PackageTableBody;
