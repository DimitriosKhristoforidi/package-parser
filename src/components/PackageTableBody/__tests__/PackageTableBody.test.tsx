import React from 'react';
import { renderWithRedux } from '../../../utils/testUtils';
import PackageTableBody from '../PackageTableBody';
import { packagesListMock } from '../../../mock/testMocks';

describe('Package Table Body', () => {
  it('Should match snapshot', () => {
    const wrapper = renderWithRedux(
      <PackageTableBody packagesList={packagesListMock} />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
