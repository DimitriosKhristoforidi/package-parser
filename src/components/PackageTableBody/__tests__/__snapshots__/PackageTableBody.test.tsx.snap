// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package Table Body Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
            scope="row"
          >
            @reduxjs/toolkit
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            The official, opinionated, batteries-included toolset for efficient Redux development
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          />
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    acemarke
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    timdorr
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    gaearon
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                  >
                    phryneas
                  </span>
                </div>
              </div>
            </div>
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
          >
            <a
              href="https://redux-toolkit.js.org"
              target="_blank"
            >
              https://redux-toolkit.js.org
            </a>
          </td>
        </tr>
      </tbody>
    </div>
  </body>,
  "container": <div>
    <tbody
      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          @reduxjs/toolkit
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          The official, opinionated, batteries-included toolset for efficient Redux development
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        />
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  acemarke
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  timdorr
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  gaearon
                </span>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-outlinedPrimary css-1t53hjd-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  phryneas
                </span>
              </div>
            </div>
          </div>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
        >
          <a
            href="https://redux-toolkit.js.org"
            target="_blank"
          >
            https://redux-toolkit.js.org
          </a>
        </td>
      </tr>
    </tbody>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
